libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,645 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ * 	- string.h
+ * TESTED:
+ * 	- strpbrk()
+ * 	- strstr()
+ * 	- strtok()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+#define ASCII_LENGTH          128
+#define EXTENDED_ASCII_LENGTH 256
+#define BUFFSIZE              24
+#define LOREM_IPSUM           "Lorem Ipsum Dolor"
+
+static char asciiStr[ASCII_LENGTH];
+static char extAsciiStr[EXTENDED_ASCII_LENGTH];
+
+TEST_GROUP(string_tok);
+TEST_GROUP(string_tok_r);
+TEST_GROUP(string_str);
+TEST_GROUP(string_pbrk);
+
+/* Ascii array creator with the ability to change the point where it will start */
+static void create_ascii_set(signed int x)
+{
+	int i;
+
+	for (i = x; i < ASCII_LENGTH; i++) {
+		asciiStr[i - x] = i;
+	}
+}
+
+
+static void create_extAscii_set(void)
+{
+	int i;
+
+	for (i = 1; i < EXTENDED_ASCII_LENGTH; i++) {
+		extAsciiStr[i - 1] = i;
+	}
+}
+
+
+TEST_SETUP(string_tok)
+{
+}
+
+
+TEST_TEAR_DOWN(string_tok)
+{
+}
+
+
+TEST(string_tok, basic)
+{
+	char *holder, dividers[4];
+
+	/* Create an Ascii array that starts from 1*/
+	create_ascii_set(1);
+
+	sprintf(dividers, "%c%c%c", asciiStr[1], asciiStr[sizeof(asciiStr) / 2], asciiStr[sizeof(asciiStr) - 1]);
+
+	/* Checking if the return of strtok is a pointer equal to the input string pointer in the first tokenization */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strtok(asciiStr, &asciiStr[sizeof(asciiStr) - 1]));
```

Comment:
I am afraid it will be necessary to assert both a pointer and a content of returned string. This way we are sure that it returned a pointer to source string in the proper position, but you don't know whether characters didn't change there.

Reply:
Above that I would check in the separate line whether in the specified position of str NUL character were placed.

Reply:
**Applies to the rest of strtok tests.**